src/org/pentaho/reporting/platform/plugin/output/CachingPageableHTMLOutput.java

```diff
@@ -307,12 +309,25 @@ private synchronized void persistContent( final String key, final IReportContent
     final IPluginCacheManager cacheManager = PentahoSystem.get( IPluginCacheManager.class );
     final IReportContentCache cache = cacheManager.getCache();
     if ( cache != null ) {
-      cache.put( key, data );
+      if ( processor.isQueryLimitReached() ) {
+        cache.put( key, data, updateQueryLimitReachedFlag( cache.getMetaData( key ) ) );
+      } else {
+        cache.put( key, data );
+      }
     } else {
       logger.error( "Plugin session cache is not available." );
     }
   }
 
+  private Map<String, Serializable> updateQueryLimitReachedFlag( Map<String, Serializable> metaData ) {
+    if ( metaData == null ) {
+      metaData = new HashMap<>();
+    }
+    metaData.put( IS_QUERY_LIMIT_REACHED, processor.isQueryLimitReached() );
+
+    return metaData;
+  }
+
```

Comment:
In case the "persistContent" method with 3 parameters is no longer used (and it seems it isn't), then please remove it. It is a private method, so it has no longer any reason to exist.